### DISCUSSION TITLE
Tools Restore MySQL file may corrupt database prompt?

### DISCUSSION THREAD

**Discussion by givihuda** [dbeaver/dbeaver#16395] [Q&A] [answered]
When I restore a mysqldump file now (since the latest update I think), I get this pop-up:

"Confirmation required to launch MySQL restore: You are about to restore database [whatever.sql]. It may corrupt your database. Are you sure?"

![image](https://user-images.githubusercontent.com/74785256/166479811-093fe558-24b3-4ee6-a516-39c8ceefcc37.png)

Why is it saying this? The sql file imports just fine.

**Comment by E1izabeth** [comment, marked as answer]
Hi, @givihuda! It's just a confirmation to make sure you understand that database will be restored and its data could be lost. We have some users who is worried about accidental usage of such dangerous operations https://github.com/dbeaver/dbeaver/issues/15068.

**Reply by givihuda** [reply]
OK, so the warning is not really accurate in its verbiage... "corrupt your database"?

Typically "corrupting" data means something got misaligned and data is now unusable...

Shouldn't the warning be something like "Your data may be overwritten based on the import" or something like that?

"corrupt your database" seems excessive and is likely inaccurate or not applicable on 99% of imports.

**Reply by E1izabeth** [reply]
Maybe you are right. Thank you for feedback.
